### PR TITLE
chore(python): Shutdown server cleanly

### PR DESCRIPTION
**Pull request by tebeka**
@efiShtain Do you have a scenario to test where the server got stuck before you added `os._exit`?

### CODE REVIEW

File: runtimes/pythonrt/runner/main.py [hunk 1]
```diff
@@ -20,9 +20,10 @@
 from grpc_reflection.v1alpha import reflection
 from syscalls import SysCalls
 import threading
-import os
 import time
 
+SERVER_GRACE = 3  # seconds
```
Comment (daabr) [inline runtimes/pythonrt/runner/main.py]: Optional nit: maybe rename or add a comment that this is a server stop/shutdown grace period? "Server grace" is a little vague.

File: runtimes/pythonrt/runner/main.py [hunk 2]
```diff
@@ -125,7 +127,7 @@ def ShouldKeepRunning(self, intial_delay=10, period=10):
             time.sleep(period)
 
         log.error("could not verify if should keep running, killing self")
-        os._exit(1)
+        self.server.stop(SERVER_GRACE)
```
Comment (daabr) [inline runtimes/pythonrt/runner/main.py]: Is it intentional that the proc's return value is no longer 1? Also, I'm assuming there's a reason you didn't use `sys.exit()`?

(Same in line 307)

**Review by daabr — APPROVED** [review]
Approved, but please reply to the comment on `self.server.stop(SERVER_GRACE)`

File: runtimes/pythonrt/runner/main.py (hunk 2 above)
Comment (tebeka) [reply runtimes/pythonrt/runner/main.py]: There's no need, server stop will cause the ` server.wait_for_termination()` to exit which will exit the process.

**Pull request merged**